curator-framework/src/main/java/org/apache/curator/framework/imps/CreateBuilderImpl.java

```diff
@@ -610,18 +610,18 @@ private String protectedPathInForeground(String adjustedPath, byte[] data, List<
         {
             ThreadUtils.checkInterrupted(e);
             if ( ( e instanceof KeeperException.ConnectionLossException ||
-                !( e instanceof KeeperException )) && protectedId != null )
+                !( e instanceof KeeperException )) && protectedMode.doProtected() )
```

Comment:
Note: this is a subtle change. The previous version checked `protectedId != null` and I don't remember why.